interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java

```diff
@@ -565,7 +564,7 @@ LoadBalancerStatsResponse waitForRpcStats(long timeoutSeconds) {
           rpcs.putAllRpcsByPeer(entry.getValue());
           builder.putRpcsByMethod(getRpcTypeString(entry.getKey()), rpcs.build());
         }
-        builder.setNumFailures(noRemotePeer + (int) latch.getCount());
+        builder.setNumFailures(rpcsFailed + (int) latch.getCount());
```

Comment:
Actually, could you clarify exactly what the problem is that you encountered with the current "wrong" semantics? (As best I can tell, these are not wrong exactly but made your new tests harder to debug, but maybe there's actually something broken here that I misunderstand). Were you getting failed RPCs that had still received a peer, so they were getting reported under rpcs_by_peer?

Since this new change still seems to group outstanding RPCs into `num_failures`, I don't see how this semantics change directly helps debug any tests with RPCs that are kept-open indefinitely by the server. Unless these end up timing out on the client and then get reported as DEADLINE_EXECEEDED? Even so, that's dependent on the parameters (client-side time out and number of seconds to wait when querying the stats service) and may not always occur. Should this change should also remove `latch.getCount()` from the reported numberOfFailures?

Reply:
None of the tests depend on distinguishing rpcs have failed vs pending on the server. This change is just trying to make  it less ambiguous. It is extremely confusing and error-prone for defining `rpcs_by_peer` to be the number of *completed RPCs* for each peers. "completed RPCs" is ambiguous here, as it includes both RPCs failed after reaching the server (e.g., DEADLINE_EXECEEDED) and RPCs finished successfully. The definition for `num_failures` is also error-prone, as its comment says it is for RPCs that failed to record a remote peer. RPCs failed after reaching the server do not account for `num_failures`, which is very surprising, isn't it?

In the new `LoadBalancerAccumulatedStatsResponse`, it clearly states the stats are for *RPCs failed* (no matter how they fails, finishing with onError() is considered as failure) and *RPCs completed successfully*. For the same batch of RPCs failed with DEADLINE_EXECEEDED, you will see `LoadBalancerStatsResponse` counts them as *completed* but `LoadBalancerAccumulatedStatsResponse` counts them as failed. 😢  

Regarding RPCs not finished in upon `GetClientStats` call times out, previously they are considered "completed" but now being "failed". Both are not satisfactory (with the latter less error-prone I think?), we can change it to not count them towards either. Still no existing tests should be affected.

Reply:
The change here is good (that's why I approved) but I think it should go further and not include the number of pending RPCs in `num_failures`. As far as I can tell, with `latch.getCount()` still included in the `num_failures` response, the change in `messages.proto`'s description of `num_failures` is not actually accurate: `num_failures` is actually being computed as the number of RPCs that failed + pending RPCs. I think we should just drop `latch.getCount()` here, and then `num_failures=Number of failed RPCs`

Reply:
Done.